src/System.Private.CoreLib/shared/System/Enum.cs

```diff
@@ -322,28 +323,29 @@ internal static ulong ToUInt64(object value)
 
         public static object Parse(Type enumType, string value, bool ignoreCase)
         {
-            bool success = TryParse(enumType, value, ignoreCase, throwOnFailure: true, out object result);
+            bool success = TryParse(enumType, value, ignoreCase, throwOnFailure: true, out object? result);
             Debug.Assert(success);
-            return result;
+            // TODO-NULLABLE: workaround compiler bug
+            return result!;
         }
 
-        public static TEnum Parse<TEnum>(string value) where TEnum : struct =>
+        public static TEnum Parse<TEnum>(string? value) where TEnum : struct =>
             Parse<TEnum>(value, ignoreCase: false);
 
-        public static TEnum Parse<TEnum>(string value, bool ignoreCase) where TEnum : struct
+        public static TEnum Parse<TEnum>(string? value, bool ignoreCase) where TEnum : struct
```

Comment:
Why do these overloads accept `string?` instead of `string`?  Don't we always throw ArgumentNullException if these are null?

Reply:
Depend how it calls `TryParse` method below, but yes it is calling with `throwOnFailure: true` so yes `value` it shouldn't be nullable